src/snowflake/snowpark/modin/plugin/__init__.py

```diff
@@ -44,13 +45,22 @@
     )  # pragma: no cover
 
 
-# We need this import here to prevent circular dependency issues, since snowflake.snowpark.modin.pandas
-# currently imports some internal utilities from snowflake.snowpark.modin.plugin. Test cases will
-# import snowflake.snowpark.modin.plugin before snowflake.snowpark.modin.pandas, so in order to prevent
-# circular dependencies from manifesting, apparently snowflake.snowpark.modin.pandas needs to
-# be imported first.
+# === INITIALIZE EXTENSION SYSTEM ===
+# Initialize all extension modules.
+import snowflake.snowpark.modin.plugin.extensions.pd_extensions  # isort: skip  # noqa: E402,F401
+import snowflake.snowpark.modin.plugin.extensions.io_overrides  # isort: skip  # noqa: E402,F401
+import snowflake.snowpark.modin.plugin.extensions.general_overrides  # isort: skip  # noqa: E402,F401
+
+# base overrides occur before subclass overrides in case subclasses override a base method
+import snowflake.snowpark.modin.plugin.extensions.base_extensions  # isort: skip  # noqa: E402,F401
+import snowflake.snowpark.modin.plugin.extensions.base_overrides  # isort: skip  # noqa: E402,F401
+import snowflake.snowpark.modin.plugin.extensions.dataframe_extensions  # isort: skip  # noqa: E402,F401
```

Comment:
@sfc-gh-joshi where did groupby go? i don't see groupby here

Reply:
We don't need to explicitly import groupby/resample/window/indexing overrides because they don't trigger any `modin.api.extensions.register_*_accessor` calls. Instead, we override the call to `DataFrame.groupby()`, `DataFrame.rolling`, and similar methods to just return a Snowpark pandas `DataFrameGroupBy`/`Window` object defined in `groupby_overrides.py`/`window_overrides.py`.